site/gatsby-site/src/components/forms/TextInputGroup.js

```diff
@@ -14,31 +14,40 @@ const TextInputGroup = ({
   handleChange,
   handleBlur,
   addOnComponent = null,
+  schema,
   className = '',
   ...props
-}) => (
-  <div className="bootstrap">
-    <Form.Group className={`form-group ${className}`}>
-      <Label popover={name} label={label} />
-      <InputGroup>
-        <Form.Control
-          type={type}
-          name={name}
-          placeholder={placeholder}
-          onChange={handleChange}
-          onBlur={handleBlur}
-          value={values[name] || ''}
-          className={touched[name] && errors[name] ? 'has-error' : null}
-          isInvalid={errors[name] && touched[name]}
-          {...props}
-        />
-        {addOnComponent}
-        <Form.Control.Feedback type="invalid">
-          <Trans ns="validation">{errors[name] && touched[name] ? errors[name] : null}</Trans>
-        </Form.Control.Feedback>
-      </InputGroup>
-    </Form.Group>
-  </div>
-);
+}) => {
+  const [optional, setOptional] = useState(true);
+
+  useEffect(async () => {
+    setOptional(await schema.fields[name].isValid(undefined));
+  }, []);
```

Comment:
Not sure I follow this 🤔

Reply:
`schema` contains the full `yup` schema. We descend into the field matching the current input group, then we test if the field is optional according to the schema by seeing if `isValid` returns `true` given `undefined`. For some reason, `isValid` is asynchronous, so we need to wrap it in a `useEffect`.

Reply:
Oh my! There must be a better way to do this, but let's go with it for now. Forms need a rework anyway, including moving them to flow bite/tailwind.